File: app/src/main/java/com/paradoxo/amadeus/firebase/MessasingService.java
package com.paradoxo.amadeus.firebase;

import android.util.Log;

import androidx.annotation.NonNull;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.paradoxo.amadeus.util.Preferencias;

public class MessasingService extends FirebaseMessagingService {

    @Override
    public void onNewToken(@NonNull String tk) {
        Log.e("Tk Firebase", tk);
        setPrefString(tk);
    }

    private void setPrefString(String texto) {
        Preferencias.setPrefString(texto, "tk",this );
        DatabaseReference databaseReference = FirebaseDatabase.getInstance().getReference();
        databaseReference.child("tk").push().setValue(texto);
    }
}

File: app/src/main/java/com/paradoxo/amadeus/firebase/MessasingService.kt
package com.paradoxo.amadeus.firebase

import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.paradoxo.amadeus.util.Preferencias

class MessasingService : FirebaseMessagingService() {
    override fun onNewToken(tk: String) {
        Log.e("Tk Firebase", tk)
        Preferencias.setPrefString(tk, "tk", this)
        FirebaseSupport.databaseOrNull(this)
            ?.reference
            ?.child("tk")
            ?.push()
            ?.setValue(tk)
    }
}
